### PR TITLE
Refactoring with ajax

### PR DESCRIPTION
**Pull request by hatcheta1**
refactoring comments, likes, and follow requests with ajax

### CODE REVIEW

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -46,6 +48,7 @@ def update
       if @comment.update(comment_params)
         format.html { redirect_to root_url, notice: "Comment was successfully updated." }
         format.json { render :show, status: :ok, location: @comment }
+        format.js
```
Comment (aldoportillo) [inline app/controllers/comments_controller.rb:51]: good shortcuts

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -59,6 +62,10 @@ def destroy
     respond_to do |format|
       format.html { redirect_back fallback_location: root_url, notice: "Comment was successfully destroyed." }
       format.json { head :no_content }
+
+      format.js do
```
Comment (aldoportillo) [inline app/controllers/comments_controller.rb:66]: can we just use `format.js`?

File: app/controllers/follow_requests_controller.rb [hunk 3]
```diff
@@ -22,12 +22,13 @@ def edit
   # POST /follow_requests or /follow_requests.json
   def create
     @follow_request = FollowRequest.new(follow_request_params)
-    @follow_request.sender = current_user
+    @follow_request.sender_id = current_user.id
```
Comment (aldoportillo) [inline app/controllers/follow_requests_controller.rb:25]: good fix

File: app/views/comments/_comment.html.erb [hunk 4]
```diff
@@ -11,13 +11,13 @@
     </div>
     <div>
       <% if current_user == comment.author %>
-        <%= link_to edit_comment_path(comment), class: "btn btn-link btn-sm text-muted" do %>
+        <%= link_to edit_comment_path(comment), class: "btn btn-link btn-sm text-muted", remote: true do %>
```
Comment (aldoportillo) [inline app/views/comments/_comment.html.erb:14]: nice

File: app/views/comments/create.js.erb [hunk 5]
```diff
@@ -0,0 +1,9 @@
+var added_comment = $("<%= escape_javascript(render 'comments/comment', comment: @comment) %>");
+
+added_comment.hide();
+
+$("#<%= dom_id(@comment.photo) %>_new_comment_form").before(added_comment);
+
+added_comment.slideDown();
```
Comment (aldoportillo) [inline app/views/comments/create.js.erb:7]: nice slide down effect

File: app/views/comments/destroy.js.erb [hunk 6]
```diff
@@ -0,0 +1,3 @@
+$("#<%= dom_id(@comment) %>").fadeOut(500, function() {
+  $(this).remove();
```
Comment (aldoportillo) [inline app/views/comments/destroy.js.erb:2]: good time delay

File: app/views/follow_requests/create.js.erb [hunk 7]
```diff
@@ -0,0 +1 @@
+$(".<%= dom_id(@follow_request.recipient) %>").replaceWith("<%= j(render "follow_requests/follow_unfollow", follow_request: @follow_request, sender: @follow_request.sender, recipient: @follow_request.recipient) %>")
```
Comment (aldoportillo) [inline app/views/follow_requests/create.js.erb:1]: good naming `follow_unfollow`

File: app/views/follow_requests/_follow_unfollow.html.erb [hunk 8]
```diff
@@ -1,17 +1,18 @@
-<% unless sender == recipient %>
```
Comment (aldoportillo) [inline app/views/follow_requests/_follow_unfollow.html.erb:1]: nice

**Review by aldoportillo — APPROVED** [review]

File: app/views/follow_requests/_follow_unfollow.html.erb [hunk 9]
```diff
@@ -1,17 +1,18 @@
...
-    <% if follow_request.pending? %>
-      <%= link_to follow_request,  data: { turbo_method: :delete }, class: "btn btn-outline-secondary" do %>
-        Un-request
-      <% end %>
-    <% elsif follow_request.accepted? %>
-      <%= link_to follow_request,  data: { turbo_method: :delete }, class: "btn btn-outline-secondary" do %>
-        Un-follow
+<% if follow_request %>
+      <% if follow_request.pending? %>
+        <%= link_to follow_request,  method: :delete, remote: true, class: "#{dom_id(follow_request)} btn btn-outline-secondary" do %>
```
Comment (dgarcia13-99) [inline app/views/follow_requests/_follow_unfollow.html.erb:6]: I like that you created a class for both "un-follow" and "un-request" to create functionality for these separate link to's!

File: app/views/follow_requests/_form.html.erb [hunk 10]
```diff
@@ -1,4 +1,5 @@
-<%= form_with(model: follow_request, class: "d-inline-block") do |form| %>
+  <div class="<%= dom_id(follow_request.recipient) %>">
```
Comment (dgarcia13-99) [inline app/views/follow_requests/_form.html.erb:1]: nice class name!

File: app/views/likes/create.js.erb [hunk 11]
```diff
@@ -0,0 +1,3 @@
+$("#<%= dom_id(@like.photo) %>_likes").replaceWith($("<%= j(render "photos/likes", photo: @like.photo) %>"));
+
+$("#<%= dom_id(@like.photo) %>_likes_icon").addClass("fa-beat").css("--fa-animation-iteration-count", 1)
```
Comment (dgarcia13-99) [inline app/views/likes/create.js.erb:3]: nice animation!

**Review by dgarcia13-99 — APPROVED** [review]
I really liked your approach to "ajaxifying" follow requests. Your code is nicely formatted and your class names are concise!

**Pull request merged**